src/backend/joanie/edx_imports/tasks/certificates.py

```diff
@@ -84,8 +87,12 @@ def import_certificates_batch(start, stop, dry_run=False):
             )
             continue
 
+        if models.Certificate.objects.filter(enrollment=enrollment).exists():
+            report["certificates"]["skipped"] += 1
+            continue
+
         organization_code, signatory = edx_mongodb.get_enrollment(
-            edx_certificate.course_id
+            enrollment.course_run.course.code
```

Comment:
line 106 you have to do this change : 

```
            organization = models.Organization.objects.get(code__iexact=organization_code)
```

otherwise sometimes the organization is not found